thunder/dynamo/compiler.py

```diff
@@ -143,5 +143,51 @@ def thunderfx(fn: Callable, /, **kwargs) -> Callable:
 
     backend = ThunderCompiler(**thunder_options)
     compiled = torch.compile(fn, backend=backend, **torch_compile_options)
-    compiled._backend = backend
-    return compiled
+
+    # We return this object instead of just the raw `compiled` Callable so that
+    # we have a place to hang the `last_traces` property.
+    class CompiledObject:
+        def __init__(self, be, func: Callable):
+            self._backend = backend
+            self._func = func
+
+        def __call__(self, *args, **kwargs):
+            return self._func(*args, **kwargs)
+
+        @property
+        def last_traces(self) -> [Trace]:
+            """
+            Get the Thunder traces for all the forward subgraphs of a ThunderFX
+            callable.
+
+            .. note:: The object must have been invoked before calling this
+                      function.
+            """
+            rv: [Trace] = []
+            for sinfo in self._backend.subgraph_infos:
+                for th_fqn in sinfo.thunder_compiled_fns:
+                    trcs = thunder.last_traces(th_fqn)
+                    if trcs is not None and trcs != []:
```

Comment:
Curious to know if there are cases where `trcs` could be None or []. I think `trcs` should always be populated.

https://github.com/Lightning-AI/lightning-thunder/blob/21d93191e6629696b9cb875fedd9dc218c58e761/thunder/__init__.py#L827-L839

Reply:
`None` is clearly handled (i'll remove that), but an empty list happens in Alexandros' benchmark scripts for one or more of the 3 models; i frequently modify that code in _nemo.py to account for this.